### PR TITLE
query extensions

### PR DESCRIPTION
**Pull request by singhal-ekansh**

### CODE REVIEW

File: app/models/product.rb [hunk 1]
```diff
@@ -31,6 +31,11 @@ class Product < ApplicationRecord
   validates_with PriceValidator, unless: -> { price.blank? || discount_price.blank? }
   validates :price, comparison: { greater_than: :discount_price, message: 'must be greater than discount price' }, unless: -> { discount_price.blank? }
   
+  # query extentions
+  
+  scope :enabled_products, -> { where(enabled: true) }
```
Comment (harsh-ag) [inline app/models/product.rb]: rename it to only `enabled`

File: app/models/order.rb [hunk 2]
```diff
@@ -19,6 +19,19 @@ class Order < ApplicationRecord
   # ...
   validates :name, :address, :email, presence: true
   validates :pay_type, inclusion: pay_types.keys
+
+  # query extensions
+
+  scope :by_date, ( Proc.new do |from, to|
```
Comment (harsh-ag) [inline app/models/order.rb]: no need to add conditional statements. use default value for arguments

**Review by harsh-ag — CHANGES_REQUESTED** [review]

File: app/models/order.rb [hunk 3]
```diff
@@ -19,6 +19,13 @@ class Order < ApplicationRecord
   # ...
   validates :name, :address, :email, presence: true
   validates :pay_type, inclusion: pay_types.keys
+
+  # query extensions
+
+  scope :by_date, ( Proc.new do |from = Time.now.midnight, to|
```
Comment (harsh-ag) [inline app/models/order.rb]: use `-> {}` inplace of Proc.new

Comment (singhal-ekansh) [reply app/models/order.rb]: done

**Review by harsh-ag — APPROVED** [review]

**Pull request closed without merging**